test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1

```diff
@@ -0,0 +1,135 @@
+Describe "Basic Send-MailMessage tests" -Tags CI {
+    function read-mail
+    {
+        Param(
+            [parameter(Mandatory=$true)]
+            [String]
+            $mailBox
+        )
+
+        $state = "init"
+        $mail = Get-Content $mailBox
+        $rv = @{}
+        foreach ($line in $mail)
+        {
+            switch ($state)
+            {
+                "init"
+                {
+                    if ($line.Length -gt 0)
+                    {
+                        $state = "headers"
+                    }
+                }
+                "headers"
+                {
+                    if ($line.StartsWith("From: "))
+                    {
+                        $rv.From = $line.Substring(6)
+                    }
+                    elseif ($line.StartsWith("To: "))
+                    {
+                        if ($rv.To -eq $null)
+                        {
+                            $rv.To = @()
+                        }
+
+                        $rv.To += $line.Substring(4)
+                    }
+                    elseif ($line.StartsWith("Subject: "))
+                    {
+                        $rv.Subject = $line.Substring(9);
+                    }
+                    elseif ($line.Length -eq 0)
+                    {
+                        $state = "body"
+                    }
+                }
+                "body"
+                {
+                    if ($line.Length -eq 0)
+                    {
+                        $state = "done"
+                        continue
+                    }
+
+                    if ($rv.Body -eq $null)
+                    {
+                        $rv.Body = @()
+                    }
+
+                    $rv.Body += $line
+                }
+            }
+        }
+
+        return $rv
+    }
+
+    BeforeAll {
+        $PesterArgs = @{ Name = "Can send mail message from user to self"}
+        $alreadyHasMail = $true
+
+        if (-not $IsLinux)
```

Comment:
On Linux, an SMTP server is not installed by default, since there are at least two to choose from, perhaps more.

I'll have to look into how to detect from PowerShell if anyone's listening on port 25.